bokehjs/src/lib/models/widgets/markup.ts

```diff
@@ -36,6 +48,39 @@ export abstract class MarkupView extends WidgetView {
     this.markup_el = div({class: clearfix, style})
     this.el.appendChild(this.markup_el)
   }
+
+  has_math_disabled() {
+    return this.model.disable_math || !this.contains_tex_string(this.model.text)
+  }
+
+  process_tex(): string {
+    if (this.provider.status !== "loaded") return this.model.text
+
+    const tex_parts = this.provider.MathJax?.find_math(this.model.text)!
+
+    return this.process_tex_parts(tex_parts)
+  }
+
+  private process_tex_parts(math_parts: MathJax.ProtoItem[]): string {
+    const {text} = this.model
+    const parts: string[] = []
+
+    math_parts.reduce((last_index = 0, math_part) => {
```

Comment:
Is this default value really needed? It seems to me that this (and some other instances of `.reduce()` at al) could be rewritten more clearly with a simple for loop.